lib/scanner/walk.go

```diff
@@ -336,21 +329,17 @@ func (w *walker) walkRegular(ctx context.Context, relPath string, info fs.FileIn
 		}
 	}
 
-	f := protocol.FileInfo{
-		Name:          relPath,
-		Type:          protocol.FileInfoTypeFile,
-		Version:       curFile.Version.Update(w.ShortID),
-		Permissions:   newMode & uint32(fs.ModePerm),
-		NoPermissions: w.IgnorePerms,
-		ModifiedS:     info.ModTime().Unix(),
-		ModifiedNs:    int32(info.ModTime().Nanosecond()),
-		ModifiedBy:    w.ShortID,
-		Size:          info.Size(),
-		RawBlockSize:  int32(blockSize),
-		LocalFlags:    w.LocalFlags,
-	}
+	f, _ := CreateFileInfo(info, relPath, nil)
+	f = w.updateFileInfo(f, curFile)
+	f.NoPermissions = w.IgnorePerms
+	f.RawBlockSize = int32(blockSize)
 
 	if hasCurFile {
+		if runtime.GOOS == "windows" {
```

Comment:
could/should this happen in w.updateFileInfo?

Reply:
I tried to keep that function simple. This would require passing `hasCurFile` and conditions on that, the file type and goos == windows. Personally I'd rather not do that, but it doesn't really matter.

Reply:
It already has curFile and the job to update the new based on the old, so it feels to me it would fit there.